Rework to MCuC model, Added all VCU fault injection, Updated all MCuC and HM datatypes

njk4401 opened the pull request:
Comments in cyan are documentation
Comments in red are questions moving forward
Comments in yellow are todo

njk4401 committed:
Added Key_Cycle state between ESTOP and MC_OFF. Removed transitions into Fault.

njk4401 committed:
Soup in the pot

njk4401 committed:
Updates to rev3_data

mwd9475 committed:
Heartbeats Added

njk4401 committed:
Merge pull request #23 from RIT-EVT/Heartbeats

Heartbeats Added

njk4401 committed:
Overhauled MCuC; Combined signals, Added Super Fault state, changed order of UC_State enum

njk4401 committed:
Insanely good organization + Basic heartbeat checking

njk4401 committed:
Minor tweaks

njk4401 committed:
Changed HB to uin32, Brake does something

njk4401 committed:
ok

dd9089 committed:
added switches to all fault injected signals in fAULT INJECTION

njk4401 committed:
Added every fault injection you could imagine

dd9089 committed:
Changed any doubles to ints

njk4401 committed:
Cleaned up

njk4401 committed:
Changed fault logic

njk4401 committed:
Added and tested all fault/error conditions

mjmagee991 commented:
> Comments in cyan are documentation Comments in red are questions moving forward Comments in yellow are todo

This would be good to document in a permanent location somewhere

mjmagee991 commented:
The title and branch name makes it sound like you only touched MCuC, but pretty much everything got modified. What did you do to all those files?

mjmagee991 commented:
Can we do something a little more subtle than this. This is huge
![image](https://github.com/user-attachments/assets/b8f5750c-6f71-4059-a440-3a891c7d0cd1)

mjmagee991 commented:
I'm realizing that on DEV1, forward enable allows you to temporarily disable the inverter but doesn't require hitting start again to re-enable. You can just flick the forward enable back and forth to enable and disable movement. Do you think that's how it should work on REV3?

mjmagee991 commented:
Why do you need this delay here? Ideally the only unit delay would be in the outer loop
![image](https://github.com/user-attachments/assets/2dafb763-3371-45f1-86a6-f702fdcb3a36)

mjmagee991 commented:
This makes sense, but I think it should just be an internal signal, so you don't have to worry about it at the top level
![image](https://github.com/user-attachments/assets/f2b466e9-f431-4a39-9c5a-ec8be179ae78)

mjmagee991 commented:
Mention why it would be pointless: this model is only used for testing the VCU, so not having power means it's useless
![image](https://github.com/user-attachments/assets/4825cfc1-e2be-4ccf-a646-57e7b3ebed6c)

mjmagee991 commented:
What do gray comment blocks mean?
![image](https://github.com/user-attachments/assets/6abb29f5-3f9e-4478-8266-7879d9ac81a4)

njk4401 commented:
> The title and branch name makes it sound like you only touched MCuC, but pretty much everything got modified. What did you do to all those files?

Everything done that wasn't in MCuC:
Added heartbeat signal to BMS, GFDB, HIB, TMS, LVSS
Updated datatypes in Hardmon
Added VCU fault injection signals to VCU block and REV3 block
Updated rider inputs interface
Rearranged UC_State enum

njk4401 committed:
Added new state between Contactor_Closed and MC_Active

njk4401 committed:
Made comment style consistent across all files, LVSS_Outputs go out of Control_Step and back thru feedback, Removed debugging lights on top level

mjmagee991 commented:
Comments in yellow are things that need _to be_ done
![image](https://github.com/user-attachments/assets/9ef4160e-bd74-4c3d-b497-6ef5576796e5)

mjmagee991 commented:
Also move this comment into the blocks
![image](https://github.com/user-attachments/assets/62d76075-a629-4ba7-9b14-f261a1e2bf22)

mjmagee991 commented:
Not a big deal, but visually, I feel like it would be nice if all the blocks were the same size. I also think it would make sense to make all the blocks smaller, so the text is relatively larger and easier to read with out zooming and panning
![image](https://github.com/user-attachments/assets/32c01388-b3c3-4fdd-b2ed-201c3e2e2e3f)

mjmagee991 commented:
Move GFDB down past BMS, so its output doesn't intersect with the VCU output line
![image](https://github.com/user-attachments/assets/e6d2858b-e202-44f4-be52-4c51dd802231)

mjmagee991 commented:
In HIB, we call the enable signal, "Enable," "EN," and "Enabled". I'm okay with either Enable or EN, but I would like to be consistent across all boards. Looks like this is an issue in pretty much every block right now
![image](https://github.com/user-attachments/assets/eb0bae5e-e16a-4e65-a60a-5f235dba9c40)

mjmagee991 commented:
None of the resettable or enabled subsystems have been named. I'm not sure what the names should be, but we should have a standard name for this for consistency
![image](https://github.com/user-attachments/assets/b7b25559-6c95-41b0-867f-9b14fa042a90)

mjmagee991 commented:
On all the other boards, this is called enable, but for the LVSS, it's called "Internal_Signal" for some reason. This should be made standard
![image](https://github.com/user-attachments/assets/0354a68d-7a14-4d1f-ab87-b28a07a63a2d)

I also just realized that LVSS has an enabled subsystem wrapping a resettable subsystem, which is the opposite of what the others do. It also has all its logic directly in the resettable subsystem instead of being wrapped in another subsystem layer. This should be made the same as the other boards

mjmagee991 commented:
This is a nitpick, but can you make the boxes the right size to make all the lines straight, and can you make the resettable subsystem less oblong? This is just unnecessary visual noise
![image](https://github.com/user-attachments/assets/3b20167d-3294-473d-8142-63b0f2e91d74)

mjmagee991 commented:
Why is this a vector?
![image](https://github.com/user-attachments/assets/6578a5ed-72dc-42a6-9148-e5efc43f94f0)

mjmagee991 commented:
It's going to be much easier for you to add signals that you don't use now than to change the test harnesses later, so I'd recommend just adding a way to override these before we go too far
![image](https://github.com/user-attachments/assets/25b4d170-2f19-444b-9c4b-e02dd9c41b63)

mjmagee991 commented:
Add a comment explaining why these are terminated
![image](https://github.com/user-attachments/assets/e1e10b62-12de-41b2-8edb-e0e537329a48)

You also might consider putting the bus expander and terminators  in a subsystem to make it a little cleaner

mjmagee991 commented:
For consistency, I think I'd prefer all the switches and all the real outputs to be down by the MCuC, so you can see it all in one place. You can use gotos to bring the signals down instead of sending them up
![image](https://github.com/user-attachments/assets/cc037028-b61f-4494-8172-a5ba9e25ddba)

mjmagee991 commented:
In my opinion, these bus creators are way too thick. Even if they're really long, I would just make them the default thickness
![image](https://github.com/user-attachments/assets/d0f9be80-d276-45e2-a81d-0e5e7e9e49e3)

mjmagee991 commented:
I don't completely hate it, but why are you organizing things like this? It seems strange to me if each signal is only being used in one place
![image](https://github.com/user-attachments/assets/5c3fe00d-0417-4d9c-8a64-674d669518f7)

mjmagee991 commented:
For right now, if anything fails a heartbeat check, I would like to throw an error and just give up. In the future, we probably want to be smarter about this
![image](https://github.com/user-attachments/assets/a9e708b4-af9a-4b8f-9672-8edb9bca58c7)

mjmagee991 commented:
What's the deal with this random delay?
![image](https://github.com/user-attachments/assets/281bd119-ade6-4e67-9297-b982a9b06518)

mjmagee991 commented:
When does this error occur? Is it possible for MCuC to catch this without the Hardmon stepping in?
![image](https://github.com/user-attachments/assets/a4c26de4-0eec-4cac-960a-73c43c34a9d3)

mjmagee991 commented:
Why have a separate category of error like this? What's the advantage of skipping discharge?
![image](https://github.com/user-attachments/assets/5f2f1313-6185-4e8b-96a8-b089d31ed9da)

mjmagee991 commented:
I'm pretty confident this is not going to work on the actual system because there's going to be a delay of more than 1 cycle, but I don't really know. We'll have to play around with it when we get to that point
![image](https://github.com/user-attachments/assets/1fd5a448-d603-4bb6-bf9c-908bc8cf7128)

If you could make this easily tuneable to different delays or to accept a range of delays, that would be great. As it stands, you'd have to know exactly how many cycles to get this to work right

mjmagee991 commented:
We're going to have to tune this watchdog parameter to make it effective. Letting the MCuC be dead for a full second without doing anything about it probably isn't great
![image](https://github.com/user-attachments/assets/8b4fa99a-883e-4467-b71f-a6741bdc4961)

mjmagee991 commented:
MCuC logic should handle the fault logic. Cooling logic should just set pump speed
![image](https://github.com/user-attachments/assets/d3d43075-e72c-406f-a6b8-d00642f5eebc)

mjmagee991 commented:
This is a good question. On one hand, ESTOP is triggered when the rider has an issue, which is presumably unrelated to the health stuff we're tracking. That means the battery should be fine. On the other hand, if the rider feels the need to hit ESTOP, there might be an issue we didn't catch that calls for opening the battery contactors
![image](https://github.com/user-attachments/assets/51db11d8-75cb-4b1b-bc2b-454cb4f3effc)

mjmagee991 commented:
What is the point of the preset state? It doesn't really make sense to me why this is necessary
![image](https://github.com/user-attachments/assets/f9669a8f-9373-4e59-aacb-46274f926360)

mjmagee991 commented:
What is "uncharged"?
![image](https://github.com/user-attachments/assets/0a304482-b87d-468c-ae7e-4623ea87e289)

mjmagee991 commented:
Our riders will kill you for this
![image](https://github.com/user-attachments/assets/d16d60e8-b252-4342-b93c-d44a61c785d5)

mjmagee991 commented:
We should talk about this tomorrow. I'm not sure whether I like this here
![image](https://github.com/user-attachments/assets/bc01bb88-e99a-4793-984a-f00af8b639b6)

njk4401 committed:
Board internal consistency

njk4401 committed:
Cleaned control step block

njk4401 committed:
Removed TODO comments that have been done

dd9089 commented:
> Why is this a vector? ![image](https://private-user-images.githubusercontent.com/54601318/434572831-6578a5ed-72dc-42a6-9148-e5efc43f94f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ4OTg1ODMsIm5iZiI6MTc0NDg5ODI4MywicGF0aCI6Ii81NDYwMTMxOC80MzQ1NzI4MzEtNjU3OGE1ZWQtNzJkYy00MmE2LTkxNDgtZTVlZmM0M2Y5NGYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE3VDEzNTgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzOTczODVhZjhjNDczN2QyNjFiNWQzZDc0MTdmNmEwYTY1ZjdlYWY4ZjI1MTI4NTlhNTZjZTA3ZmQ3MmQwYTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hu_8p5TjzFJxqNUdeRdcqLAtSEwP9oyYoXpunb3y34c)

We made this a vector so that we wouldnt need to deal with all the signals. It makes it look so much cleaner.

dd9089 commented:
> What's the deal with this random delay? ![image](https://private-user-images.githubusercontent.com/54601318/434577785-281bd119-ade6-4e67-9297-b982a9b06518.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ4OTg1ODMsIm5iZiI6MTc0NDg5ODI4MywicGF0aCI6Ii81NDYwMTMxOC80MzQ1Nzc3ODUtMjgxYmQxMTktYWRlNi00ZTY3LTkyOTctYjk4MmE5YjA2NTE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE3VDEzNTgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZiMDI0ZTgzNjg0MDdiZmJhNzYzNmY0MjU2NmJiYWRmYWMxMGY3MDNlYWNhMGFhYzVhMTdhNzMwODg2Yzg1MWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UoHsJpbwVJBfh17HqAlJPFW34H6FlCWev4SWiBkmxy4)

since were taking the other signal from the feedback, theres a delay so we needed to match it by adding another delay

dd9089 commented:
> I'm pretty confident this is not going to work on the actual system because there's going to be a delay of more than 1 cycle, but I don't really know. We'll have to play around with it when we get to that point ![image](https://private-user-images.githubusercontent.com/54601318/434580908-1fd5a448-d603-4bb6-bf9c-908bc8cf7128.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ4OTg1ODMsIm5iZiI6MTc0NDg5ODI4MywicGF0aCI6Ii81NDYwMTMxOC80MzQ1ODA5MDgtMWZkNWE0NDgtZDYwMy00YmI2LWJmOWMtOTA4YmM4Y2Y3MTI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE3VDEzNTgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlMzg1OWI4ZjM1N2Y5NzhhMjM2MTBlYjUxMTdmOTA5MjFlNDg0ZjAyM2JmODMyMzIyMjI0NmVmNjEyMjE0ZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6vpyoo6DvmqEZet8Y9-yGqUxRVVtYuho-HQz3B37S2g)
> 
> If you could make this easily tuneable to different delays or to accept a range of delays, that would be great. As it stands, you'd have to know exactly how many cycles to get this to work right

would making the cycle delay be determined by a variable in the workspace be good enough? youd be able to change it at before compiling and we could make it so the variable is always set and created before compiling. Im not sure if you could link a variable to a constant block but ill do some research.

dd9089 commented:
> > I'm pretty confident this is not going to work on the actual system because there's going to be a delay of more than 1 cycle, but I don't really know. We'll have to play around with it when we get to that point ![image](https://private-user-images.githubusercontent.com/54601318/434580908-1fd5a448-d603-4bb6-bf9c-908bc8cf7128.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ4OTg1ODMsIm5iZiI6MTc0NDg5ODI4MywicGF0aCI6Ii81NDYwMTMxOC80MzQ1ODA5MDgtMWZkNWE0NDgtZDYwMy00YmI2LWJmOWMtOTA4YmM4Y2Y3MTI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE3VDEzNTgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlMzg1OWI4ZjM1N2Y5NzhhMjM2MTBlYjUxMTdmOTA5MjFlNDg0ZjAyM2JmODMyMzIyMjI0NmVmNjEyMjE0ZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6vpyoo6DvmqEZet8Y9-yGqUxRVVtYuho-HQz3B37S2g)
> > If you could make this easily tuneable to different delays or to accept a range of delays, that would be great. As it stands, you'd have to know exactly how many cycles to get this to work right
> 
> would making the cycle delay be determined by a variable in the workspace be good enough? youd be able to change it at before compiling and we could make it so the variable is always set and created before compiling. Im not sure if you could link a variable to a constant block but ill do some research.

never mind you can make the delay blocks take the number of cycles as an input

dd9089 commented:
> Our riders will kill you for this ![image](https://private-user-images.githubusercontent.com/54601318/434585201-d16d60e8-b252-4342-b93c-d44a61c785d5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ4OTg1ODMsIm5iZiI6MTc0NDg5ODI4MywicGF0aCI6Ii81NDYwMTMxOC80MzQ1ODUyMDEtZDE2ZDYwZTgtYjI1Mi00MzQyLWI5M2MtZDQ0YTYxYzc4NWQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE3VDEzNTgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEyYWQ3N2I3N2NmMWRkZTQzNmUxZmYzYmRiZTliNGYwOWM5ZTI2MmM5NTNkOTUzMTEyMjNhNmI0NGNkMmFjMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.OKVj38ha9eGnnXlpStz4LJeQ9nT2atbaFoDKFjGuzWw)

it makes it look so nice

dd9089 commented:
> When would these transitions occur? ![image](https://private-user-images.githubusercontent.com/54601318/434585899-759bb456-ea82-43e6-be55-296123546e07.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ4OTg1ODMsIm5iZiI6MTc0NDg5ODI4MywicGF0aCI6Ii81NDYwMTMxOC80MzQ1ODU4OTktNzU5YmI0NTYtZWE4Mi00M2U2LWJlNTUtMjk2MTIzNTQ2ZTA3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE3VDEzNTgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4MjdiMzZmYTk4MGU1NTZkZWUzNmJkYTA1ZGMwMDBhMDQ2N2UyMTNmMjdmNGQzM2FjMDkyMmRhNmYyMjBmYTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5IQUllr_gk4rsIAhFQrg6W0WQ9AO1b6XhB14U0E87Y0)

They occur whenever the fault is triggered and the model hasnt gone through discharge, this was done as a way to make the faults go through discharge first

njk4401 committed:
Added unused signals to fault injection

njk4401 committed:
Table 3 is still waiting

mwd9475 committed:
Renamed Reset and Enable subsystems

njk4401 committed:
Moved switches in VCU

njk4401 committed:
Fixed Matthew's problems

njk4401 committed:
*some

dd9089 commented:
> When does this error occur? Is it possible for MCuC to catch this without the Hardmon stepping in? ![image](https://private-user-images.githubusercontent.com/54601318/434578440-a4c26de4-0eec-4cac-960a-73c43c34a9d3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5MzY3NjMsIm5iZiI6MTc0NDkzNjQ2MywicGF0aCI6Ii81NDYwMTMxOC80MzQ1Nzg0NDAtYTRjMjZkZTQtMGVlYy00Y2FjLTk2MGEtNzNjNDNjMzRhOWQzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE4VDAwMzQyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAwZDVhY2I1NjVjNjVmN2U1MDBmM2VlNjAyYTY1YWUzOThmNmQ5ZmQ5NmUwNTMzZGQ4Y2IwM2NlZDUxODViZGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Q9llD3YtTis82v4oviPhjpCPTmal2_eVl2EcbmJPk5w)

NO!!!! Please dont ask any more of these types of questions

dd9089 commented:
> What is the point of the preset state? It doesn't really make sense to me why this is necessary ![image](https://private-user-images.githubusercontent.com/54601318/434583490-f9669a8f-9373-4e59-aacb-46274f926360.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5MzY3NjMsIm5iZiI6MTc0NDkzNjQ2MywicGF0aCI6Ii81NDYwMTMxOC80MzQ1ODM0OTAtZjk2NjlhOGYtOTM3My00ZTU5LWFhY2ItNDYyNzRmOTI2MzYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE4VDAwMzQyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViOGVhNGQwOGQ1Mjk2MTEwYTRiOTEzOTdkNzQ1ZDg5MGVlOGJiY2I0NDdhMWNlMmZlNGZmN2Y4MzcyNTJhNDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.bRxq-BJ5VUe4PfeiY-UE6cVY3NK62AwBlYbokYtEyuM)

Preset will mostly be used for the self testing. So right now we dont really use it, but it will be used for running the self tests.

njk4401 committed:
Soup is a lost cause

njk4401 committed:
The soup spilled...

mwd9475 committed:
The soup has been cleaned

njk4401 committed:
Changed rider input display, added semicolons where needed so nothing printed to console when running

mjmagee991 commented:
> What's the deal

This is going to be a disaster, but we can merge and deal with it later

mjmagee991 commented:
> > I'm pretty confident this is not going to work on the actual system because there's going to be a delay of more than 1 cycle, but I don't really know. We'll have to play around with it when we get to that point ![image](https://private-user-images.githubusercontent.com/54601318/434580908-1fd5a448-d603-4bb6-bf9c-908bc8cf7128.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ4OTg1ODMsIm5iZiI6MTc0NDg5ODI4MywicGF0aCI6Ii81NDYwMTMxOC80MzQ1ODA5MDgtMWZkNWE0NDgtZDYwMy00YmI2LWJmOWMtOTA4YmM4Y2Y3MTI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE3VDEzNTgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlMzg1OWI4ZjM1N2Y5NzhhMjM2MTBlYjUxMTdmOTA5MjFlNDg0ZjAyM2JmODMyMzIyMjI0NmVmNjEyMjE0ZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6vpyoo6DvmqEZet8Y9-yGqUxRVVtYuho-HQz3B37S2g)
> > If you could make this easily tuneable to different delays or to accept a range of delays, that would be great. As it stands, you'd have to know exactly how many cycles to get this to work right
> 
> would making the cycle delay be determined by a variable in the workspace be good enough? youd be able to change it at before compiling and we could make it so the variable is always set and created before compiling. Im not sure if you could link a variable to a constant block but ill do some research.

This is not going to work. We need to allow the signal to be late by a range of delays. We aren't going to get a consistent delay that we can just tune to

mjmagee991 commented:
> > When would these transitions occur? ![image](https://private-user-images.githubusercontent.com/54601318/434585899-759bb456-ea82-43e6-be55-296123546e07.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ4OTg1ODMsIm5iZiI6MTc0NDg5ODI4MywicGF0aCI6Ii81NDYwMTMxOC80MzQ1ODU4OTktNzU5YmI0NTYtZWE4Mi00M2U2LWJlNTUtMjk2MTIzNTQ2ZTA3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE3VDEzNTgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4MjdiMzZmYTk4MGU1NTZkZWUzNmJkYTA1ZGMwMDBhMDQ2N2UyMTNmMjdmNGQzM2FjMDkyMmRhNmYyMjBmYTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5IQUllr_gk4rsIAhFQrg6W0WQ9AO1b6XhB14U0E87Y0)
> 
> They occur whenever the fault is triggered and the model hasnt gone through discharge, this was done as a way to make the faults go through discharge first

Add a comment explaining what uncharged is

mjmagee991 commented:
Kill super super fault

mjmagee991 commented:
> > What is the point of the preset state? It doesn't really make sense to me why this is necessary ![image](https://private-user-images.githubusercontent.com/54601318/434583490-f9669a8f-9373-4e59-aacb-46274f926360.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5MzY3NjMsIm5iZiI6MTc0NDkzNjQ2MywicGF0aCI6Ii81NDYwMTMxOC80MzQ1ODM0OTAtZjk2NjlhOGYtOTM3My00ZTU5LWFhY2ItNDYyNzRmOTI2MzYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE4VDAwMzQyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViOGVhNGQwOGQ1Mjk2MTEwYTRiOTEzOTdkNzQ1ZDg5MGVlOGJiY2I0NDdhMWNlMmZlNGZmN2Y4MzcyNTJhNDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.bRxq-BJ5VUe4PfeiY-UE6cVY3NK62AwBlYbokYtEyuM)
> 
> Preset will mostly be used for the self testing. So right now we dont really use it, but it will be used for running the self tests.

Call it self test then

njk4401 committed:
Feedback changes

mwd9475 committed:
Do we have bread sticks?

njk4401 committed:
Stupid stuff

njk4401 committed:
more stupid stuff

njk4401 committed:
Reverted what matt did

njk4401 committed:
Resized a lot of blocks, Pressed Space inside of every subsystem

dd9089 committed:
condesed IO into a bus for mcuc
still need to figure out error and do the same for hardmon

mwd9475 committed:
Changed I/O of Hardmon to Busses

dd9089 committed:
Fixed CAN bus switch issue
connected all signals

njk4401 committed:
Merge pull request #25 from RIT-EVT/signalbus

Signalbus

njk4401 committed:
Removed irrelevant AND gate for HUDL LEDs

mwd9475 committed:
Fixed some naming in BMS

njk4401 committed:
Added override signals for those that didn't exist
Made internal VCU signals go through the internal feedback bus

njk4401 committed:
Pressed space inside every subsystem
Removed signals taken out of VCU v2

mwd9475 reviewed (approved):
Hardmon needs to be overhaled.
Test harnessing for MCuc, Hardmon, Top level, VCU control step.
Code gen for model
